### PR TITLE
Feat(Orderportal): Support new RML OF

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
## Description

Support new orderform described in AM document 1604:13

### Added
- Support for the new orderform described in AM document 1604:1

### How to prepare for test
- [x] ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] paxa the environment: `paxa`

---

### How to test new upgrade script
- [x] do upgrade stage on Hasta

### Expected test outcome
- [x] check that all old nipt-viewer data-delivery has been converted to statina 

---

### How to test new excel orderform 
- [x] do upload the new orderform

### Expected test outcome
- [x] check that the data-analysis is selected correctly
  - [x] RML (No-analysis)
  - [x] Fluffy
- [x] check that the data-delivery is selected correctly
  - [x] fastq
  - [x] Statina
- [x] check that the control is selected correctly

---

### How to test old json orderform 
- [x] do upload the old orderform

### Expected test outcome
- [x] check that the data-analysis is selected correctly
  - [x] Fluffy
- [x] check that the data-delivery is selected correctly
  - [x] Statina

---

- [x] Take a screenshot and attach or copy/paste the output.

## Review
- [x] code approved by HS
- [x] tests executed by PG
- [x] "Merge and deploy" approved by PG
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Deploy this branch to Hasta
- [ ] Deploy this branch to cgvs-prod
- [ ] Deploy cgweb [branch](https://github.com/Clinical-Genomics/cgweb/pull/360)
- [x] Publish OF to SuSy
- [ ] Inform to AG, AZ

### CODE REVIEW

**Review by henrikstranneheim — APPROVED** [review]
Nice!

File: cg/apps/orderform/excel_orderform_parser.py [hunk 1]
```diff
@@ -158,8 +148,18 @@ def get_project_type(self, document_title: str) -> str:
             if document_number in document_title:
                 return value
 
+        analysis = self.parse_data_analysis()
+        if "1604" in document_title:
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/apps/orderform/excel_orderform_parser.py]: fixed with an enum

File: cg/apps/orderform/excel_orderform_parser.py [hunk 2]
```diff
@@ -158,8 +148,18 @@ def get_project_type(self, document_title: str) -> str:
...
+            if analysis == self.NO_ANALYSIS:
+                return str(OrderType.RML)
+            else:
+                return analysis
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/apps/orderform/excel_orderform_parser.py]: removed

File: cg/apps/orderform/excel_orderform_parser.py [hunk 3]
```diff
@@ -158,8 +148,18 @@ def get_project_type(self, document_title: str) -> str:
...
+
         if "1508" in document_title:
-            return self.get_project_type_from_samples()
+            if analysis == self.NO_ANALYSIS:
+                return str(OrderType.FASTQ)
+            else:
+                return analysis
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/apps/orderform/excel_orderform_parser.py]: removed

File: cg/apps/orderform/excel_orderform_parser.py [hunk 4]
```diff
@@ -176,24 +176,17 @@ def is_from_orderform_without_data_delivery(self, data_delivery: str) -> bool:
...
-
-        if project_type == OrderType.MICROSALT:
+        elif project_type == OrderType.MICROSALT:
             data_analysis: str = self.parse_data_analysis()
 
             if data_analysis == "custom":
                 return DataDelivery.FASTQ_QC
-
-            if data_analysis == "fastq":
+            elif data_analysis == "fastq":
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/apps/orderform/excel_orderform_parser.py:187]: We'll use in the next update of the OF

**Pull request merged**